BrainPortal/app/controllers/sessions_controller.rb

```diff
@@ -51,8 +51,16 @@ def new #:nodoc:
   end
 
   def create #:nodoc:
-    user = User.authenticate(params[:login], params[:password]) # can be nil if it fails
-    create_from_user(user)
+    # If the csrf token is blank, it was reset by Rails' request forgery protection
+    # and we want to prevent login
+    if current_session["_csrf_token"]
```

Comment:
That doesn't seem right, this test only verify that the token exists... ? Let me try it first...